### PR TITLE
General Cleanup + Switch to plugin system

### PR DESCRIPTION
**Pull request by Jeff-Ciesielski**
While not a garbage fire, this code needs lots of TLC, so if everyone is ok with it, I'd rather not bikeshed about small syntactic quibbles (There are plenty of them lying about), and just sort of ram a few of these cleanups through.  I plan on doing a full PEP8 sweep and a bunch of smaller cleanups.  I'm hoping to do a full scale review at the end.

@keyme/robotics-engineers

### CODE REVIEW

File: XBVC/xbvcgen.py [hunk 1]
```diff
@@ -1,79 +1,70 @@
...
-    messages = data.get('messages', {})
 
-    for enum_name, enum_content in enums.items():
-        m = Enum(enum_content, enum_name)
-        cs.add_member(m)
+# Adapted from https://stackoverflow.com/questions/1057431
+def _load_emitters():
+    result = {}
+    for importer, package_name, _ in pkgutil.iter_modules([EMITTER_DIR]):
+        full_package_name = '{}'.format(package_name)
```
Comment (shonfeder) [inline XBVC/xbvcgen.py]: What's this line doing?

Comment (Jeff-Ciesielski) [reply XBVC/xbvcgen.py]: seemingly nothing of consequence

File: XBVC/emitters/EmitterBase.py [hunk 2]
```diff
@@ -17,7 +17,7 @@ def __str__(self):
 
 # Base emitter class, doesn't do much, just provides stubs for child
 # classes and provides a convenient way to expand a template
-class Emitter(object):
+class EmitterBase(object):
```
Comment (knitHacker) [inline XBVC/emitters/EmitterBase.py]: remove the (object)

**Comment by Jeff-Ciesielski** [comment]
Feedback addressed

**Pull request merged**